linode/manager
# refactor: [M3-7437] - Prepare Profile Queries for React Query v5

## Description 📝

Changes of plans: No feature branch for the React Query update project. We will do all of the required changes first (merging to develop) then bump the React Query version **last**. This will eliminate the need for a feature branch.

## Changes  🔄
- This PR updates our profile related queries to 
  - Use a Query Key Factory 🏭 to help with type safety and organization 🗂️
    - Also enforces query keys are `Array` (required for React Query v4)
  - Use a single object when defining the react query hook 1️⃣ (required for React Query v5)

## How to test 🧪

> [!important]
> These code changes should result in no changes to the functionality/behavior of Cloud Manager. It is simply a migration to newer patterns.

### Verification steps 
- Check out this PR
- Test general functionality of Profile related features
  - Verify you profile loads
  - Verify grants load for restricted users
  - Verify your SSH keys load
  - Verify your App tokens load
  - Verify your PATs load

## As an Author I have considered 🤔
